### PR TITLE
WIP tracing support

### PR DESCRIPTION
**Pull request by david-driscoll**
Added [tracing support](https://www.w3.org/TR/trace-context/#header-name) with help from StreamJson RPC

### CODE REVIEW

**Comment by NTaylorMullen** [comment]
Very cool! It's nice to see under the hood and see how this stuff stitches itself together

File: src/Dap.Protocol/DapReceiver.cs [hunk 1]
```diff
@@ -81,16 +83,29 @@ protected virtual IEnumerable<Renor> GetRenor(JToken @object)
...
+                        };
+                        yield break;
+                    }
+                }
 
-                    yield return new Request(sequence, RequestNames.Cancel, ro);
+                {
+                    yield return new Request(sequence, requestName, requestObject) {
+                        TraceState = request.TryGetValue("tracestate", out var ts) ? ts.Value<string>() : null,
+                        TraceParent = request.TryGetValue("traceparent", out var tp) ? tp.Value<string>() : null,
```
Comment (AArnott) [inline src/Dap.Protocol/DapReceiver.cs:105]: Per [the spec](https://www.w3.org/TR/trace-context/#header-name), this must be a case insensitive header name match. Is that how `request.TryGetValue` is implemented?

File: src/Dap.Protocol/DebugAdapterConverters/DapClientNotificationConverter.cs [hunk 2]
```diff
@@ -33,6 +33,16 @@ public override void WriteJson(JsonWriter writer, OutgoingNotification value, Js
                 writer.WritePropertyName("body");
                 serializer.Serialize(writer, value.Params);
             }
+            if (value.TraceParent != null)
+            {
+                writer.WritePropertyName("traceparent");
+                writer.WriteValue(value.TraceParent);
+            }
+            if (value.TraceState != null)
```
Comment (AArnott) [inline src/Dap.Protocol/DebugAdapterConverters/DapClientNotificationConverter.cs:41]: [The spec](https://www.w3.org/TR/trace-context/#tracestate-header-field-values) discourages sending the `tracestate` header with an empty value. Should this condition be changed to `!string.IsNullOrEmpty(value.TraceState)`?

Comment (AArnott) [reply src/Dap.Protocol/DebugAdapterConverters/DapClientNotificationConverter.cs:41]: Also, this block should be *inside* the `if (value.TraceParent != null)` block to satisfy [the spec](https://www.w3.org/TR/trace-context/#no-traceparent-received) that says `tracestate` without `traceparent` is invalid.

**Review by AArnott — APPROVED** [review]
Looks good. I'm glad the vs-streamjsonrpc code helped expedite this.

File: src/Dap.Protocol/DapReceiver.cs (hunk 1 above)
Comment (david-driscoll) [reply src/Dap.Protocol/DapReceiver.cs:105]: Thanks!

File: src/Dap.Protocol/DebugAdapterConverters/DapClientNotificationConverter.cs (hunk 2 above)
Comment (david-driscoll) [reply src/Dap.Protocol/DebugAdapterConverters/DapClientNotificationConverter.cs:41]: Thanks!

**Comment by AArnott** [comment]
If I had $1 for every PR that completed with "WIP" in the title...

**Pull request merged**